client/client-multi/client-cpp/impl/event/EventTransport.cpp

```diff
@@ -53,44 +51,23 @@ std::shared_ptr<EventSyncRequest> EventTransport::createEventRequest(std::int32_
         }
         request->eventListenersRequests.set_array(requests);
     }
-
     if (isEventSNSynchronized_) {
-        auto pendingEvents(eventDataProcessor_.releasePendingEvents());
-        KAA_MUTEX_UNIQUE_DECLARE(lock, eventsGuard_);
-        for (auto it = events_.begin(); it != events_.end(); ++it) {
-            pendingEvents.insert(pendingEvents.end(), it->second.begin(), it->second.end());
-        }
-        events_.clear();
-
-        if (pendingEvents.empty()) {
-            request->events.set_null();
-        } else {
-            std::vector<Event> eventsCopy;
-            for (const Event& e : pendingEvents) {
-                eventsCopy.push_back(e);
-            }
-            std::sort(eventsCopy.begin(), eventsCopy.end(),
-                    [&](const Event& l, const Event& r) -> bool { return l.seqNum < r.seqNum; });
-
-            if (eventsCopy.begin()->seqNum != startEventSN_) {
-                if (clientStatus_) {
-                    clientStatus_->setEventSequenceNumber(startEventSN_ + eventsCopy.size());
-                }
-
-                KAA_LOG_INFO(boost::format("Put in order event sequence numbers "
-                        "(expected: %li, actual: %li)") % startEventSN_ % eventsCopy.begin()->seqNum);
-
-                for (auto& e : eventsCopy) {
-                    e.seqNum = startEventSN_++;
-                }
-            } else {
-                startEventSN_ += eventsCopy.size();
+        auto releasedEvents(eventDataProcessor_.releasePendingEvents());
+        if(releasedEvents.size()!=0) {
+            auto SN =clientStatus_->getEventSequenceNumber();
+            for(auto& Pair_ : releasedEvents) {
+                Pair_.second.seqNum = SN++;
+                clientStatus_->setEventSequenceNumber(SN);
```

Comment:
I guess it will be better to store the sequence number just once after all iterations. There is no need to do it in a loop.